### PR TITLE
Base: Adding stiffness units

### PR DESCRIPTION
**Pull request by Helios113**
Added stiffness units to FreeCAD and updated lexer

### CODE REVIEW

**Review by AjinkyaDahale — COMMENTED** [review]
The commit message is too vague. Remember that once merged, only the commit message will be visible. So "squashed commits" won't tell as much as "added stiffness units" or something to that accord would.

**Comment by Helios113** [comment]
Added new commit message.

**Comment by berndhahnebach** [comment]
https://github.com/FreeCAD/FreeCAD/pull/4913

**Comment by phopfgartner** [comment]
In src/Base/Unit.cpp, the line
    if(*this == Unit::Stiffness                   )       return QString::fromLatin1("Power");
looks like it contains a copy paste error.

File: src/App/FreeCADInit.py [hunk 1]
```diff
@@ -707,6 +707,11 @@ def report(self,msg,func,*args,**kargs):
 App.Units.KiloNewton    = App.Units.Quantity('kN')
 App.Units.MegaNewton    = App.Units.Quantity('MN')
 
+App.Units.NewtonPerMeter   = App.Units.Quantity('N/m')
+App.Units.MilliNewtonPerMeter   = App.Units.Quantity('mN/m')
+App.Units.KiloNewtonPerMeter   = App.Units.Quantity('kN/m')
+App.Units.MegaNewtonPerMeter   = App.Units.Quantity('MN/m')
```
Comment (luzpaz) [inline src/App/FreeCADInit.py:713]: please stack the `=` in a column to be uniform like the code around it

File: src/Base/Quantity.cpp [hunk 2]
```diff
@@ -314,6 +314,11 @@ Quantity Quantity::MilliNewton      (1.0           ,Unit(1,1,-2));
 Quantity Quantity::KiloNewton       (1e+6          ,Unit(1,1,-2));
 Quantity Quantity::MegaNewton       (1e+9          ,Unit(1,1,-2));
 
+Quantity Quantity::NewtonPerMeter   (1.00          ,Unit(0,1,-2)); //Newton per meter (N/m or kg/s^2)
+Quantity Quantity::MilliNewtonPerMeter   (1e-3          ,Unit(0,1,-2));
+Quantity Quantity::KiloNewtonPerMeter   (1e3          ,Unit(0,1,-2)); 
+Quantity Quantity::MegaNewtonPerMeter   (1e6          ,Unit(0,1,-2));
```
Comment (luzpaz) [inline src/Base/Quantity.cpp:320]: Please line up the first parethesis in each line over each other + the same the `,Unit`

File: src/Base/Unit.cpp [hunk 3]
```diff
@@ -454,6 +454,7 @@ QString Unit::getTypeString(void) const
     if(*this == Unit::Force                       )       return QString::fromLatin1("Force");
     if(*this == Unit::Work                        )       return QString::fromLatin1("Work");
     if(*this == Unit::Power                       )       return QString::fromLatin1("Power");
+    if(*this == Unit::Stiffness                   )       return QString::fromLatin1("Power");
```
Comment (luzpaz) [inline src/Base/Unit.cpp:457]: Probably should be `Stiffness` right?

File: src/Base/UnitsSchemaImperial1.cpp [hunk 4]
```diff
@@ -132,6 +132,11 @@ QString UnitsSchemaImperial1::schemaTranslate(const Quantity &quant, double &fac
             factor = 6.894744825494;
         }
     }
+    else if (unit == Unit::Stiffness) { //Make better, a
```
Comment (luzpaz) [inline src/Base/UnitsSchemaImperial1.cpp:135]: Comment is incomplete

**Review by luzpaz — CHANGES_REQUESTED** [review]

File: src/Base/Unit.cpp (hunk 3 above)
Comment (luzpaz) [reply src/Base/Unit.cpp:457]: Per https://github.com/FreeCAD/FreeCAD/pull/4963#issuecomment-898278264

**Comment by Helios113** [comment]
Changes applied, I hope it squashed correctly.

**Comment by luzpaz** [comment]
Thanks @Helios113 for the PR and your work on it. Follow up question: there are lots of changes in `src/Base/QuantityLexer.c` was this in the previous PR as well? (I don't recall it but I may be mistaken)

@berndhahnebach @donovaly @phopfgartner please weigh-in at your convenience

**Comment by Helios113** [comment]
It was not in originally in the previous commit. It was brought up in #4913, I made the changes and then the rebase messed up the commit history, so this is the first PR where you can see it.

There are a lot of changes there but the whole file was generated with flex. Hence, the changes were automatically applied. As far as I could see, the changes are mostly incrementing numbers and adding the stiffness information.

**Comment by donovaly** [comment]
> Thanks @Helios113 for the PR and your work on it. Follow up question: there are lots of changes in `src/Base/QuantityLexer.c`

That is correct. This file is auto-generated by flew, see my comment: https://github.com/FreeCAD/FreeCAD/pull/4913#pullrequestreview-724857604

**Review by donovaly — APPROVED** [review]
From my point of view the PR is ready to be merged. I am currently traveling abroad and can therefore not check if everything compiles. If it does, from the code perspective everything is fine and my requested changes/additions were applied correctly.

**Comment by luzpaz** [comment]
@berndhahnebach please review, thx!

**Comment by yorikvanhavre** [comment]
Ok looks good to me too. Ahoy!

**Pull request merged**